### DISCUSSION TITLE
Consulta de NF-e em lote

### DISCUSSION THREAD

**Discussion by CalebePrates** [TadaSoftware/PyNFe#277] [General]
Queria saber se existe a possibilidade de consultar NF-e em lote atualmente com essa bibliote. Procurei sobre o tópico na wiki e não achei sobre.

**Comment by CalebePrates** [comment]
Se sim, teria um exemplo? O arquivo com todas as chaves de notas que desejo consultar deve ser feito em qual extensão? Qual o tipo do retorno? Se puder clarear essas dúvidas

**Comment by juniortada** [comment]
@CalebePrates é só você colocar as consultas em um laço de repetição:
```python
chaves_nfe = ['chave_1', 'chave_2', 'etc']
con = ComunicacaoSefaz(uf, certificado, senha, homologacao)
for chave in chaves_nfe:
  envio = con.consulta_nota('nfe', chave)
  print (envio.text.encode('utf-8'))
```

**Reply by CalebePrates** [reply]
Mas nesse caso, não dá block pela quantidade de consultas?

Vamos colocar um cenário hipotético: tenho 100 notas para consultar. Realizando as consultas no loop sou bloqueado depois de 60 requisições. A receita pede algum "pause" entre uma e outra? Tem algum método que seja mais agradável para evitar bloqueios?

**Reply by juniortada** [reply]
@CalebePrates consultar muitas notas pela chave de nota não é recomendado pela SEFAZ.
Sua aplicação deve monitorar o próprio uso das consultas. O limite é de 20 consultar por hora (varia de SEFAZ autorizador).

"As consultas por chave de acesso e por NSU foram construídas para permitirem ao usuário buscar pontualmente alguma NF-e e de um período retroativo máximo de 90 (noventa) dias. Se o usuário necessitar buscar todas as NF-es de no máximo 90 dias de um CNPJ ator interessado, deve ser usada a consulta "distNSU". "

Recomendo que leia este documento:
https://www.nfe.fazenda.gov.br/portal/informe.aspx?ehCTG=false&Informe=0cu/yBLKrCs=

**Reply by CalebePrates** [reply]
Compreendo. No caso estou utilizando a biblioteca por ser uma solução pro meu problema de forma gratuita. Se eu necessitar fazer muitas consultas por dia, o recomendado seria API's pagas?

**Reply by juniortada** [reply]
O bloqueio é feito pela SEFAZ, a biblioteca não tem relação com isso. Mesmo uma biblioteca paga você terá que monitorar o seu consumo.
